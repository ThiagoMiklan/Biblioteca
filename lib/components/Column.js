"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: "string"
};

var Column = function Column(props) {
  (0, _type_validations.validate)(props, props_obj, "Column");
  var definition = props.definition == undefined ? "column" : "column " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.children);
};

var _default = Column;
exports.default = _default;