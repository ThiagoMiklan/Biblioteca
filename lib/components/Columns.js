"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: 'string'
};

var Columns = function Columns(props) {
  (0, _type_validations.validate)(props, props_obj, "Columns");
  var definition = props.definition == undefined ? "columns" : "columns " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, React.Children.map(props.children, function (child, i) {
    return child;
  }));
};

var _default = Columns;
exports["default"] = _default;