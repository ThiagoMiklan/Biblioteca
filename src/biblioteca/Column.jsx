import React from 'react';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import PropTypes from 'prop-types';
/**
 * Componente inspirado na classe column do bulma
 * Utilizado como filho do componente Columns
 * Possui filhos, pois funciona como um container para outros
 * elementos
 * Sem eventos
 */

const Column = (props) => {
    var definition = classnames(getClassName("column "+ props.definition, "Column"));
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    // caracter√≠sticas do bulma para a coluna
    definition : PropTypes.string
}

export default Column;