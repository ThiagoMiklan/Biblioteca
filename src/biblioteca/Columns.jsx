import React from 'react';
import getClassName from '../tools/getClassName';
import classnames from 'classnames';
import PropTypes from 'prop-types';

/**
 *  Funciona como um container para Column
 *  Baseado na classe Columns do Bulma
 *  Sem eventos
 */

const Columns = (props)=>{
    var definition = classnames(getClassName("columns "+ props.definition,"Columns"));

    return <div className={definition}>
        {
            React.Children.map(props.children, (child, i)=>{
                return child;
            })
        }
    </div>

}

Columns.propTypes = {
    // Caracter√≠sticas do Bulma
    definition: PropTypes.string
}



export default Columns