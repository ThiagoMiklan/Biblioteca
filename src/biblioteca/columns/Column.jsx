import React from 'react';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';
import PropTypes from 'prop-types';

const Column = (props) => {
    var definition = classnames(getClassName("column "+ props.definition, "Column"));
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    definition : PropTypes.string
}

export default Column;