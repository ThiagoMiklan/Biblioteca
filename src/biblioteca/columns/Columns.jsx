import React from 'react';
import getClassName from '../../tools/getClassName';
import classnames from 'classnames';


const Columns = (props)=>{
    var definition = classnames(getClassName("columns "+ props.definition,"Columns"));

    return <div className={definition}>
        {
            React.Children.map(props.children, (child, i)=>{
                return child;
            })
        }
    </div>

}

export default Columns