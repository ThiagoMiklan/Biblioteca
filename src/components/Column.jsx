// @flow
import * as  React from 'react';
import PropTypes from 'prop-types';
import {validate} from '../tools/type_validations.js';
/**
 * Componente inspirado na classe column do bulma
 * Utilizado como filho do componente Columns
 * Possui filhos, pois funciona como um container para outros
 * elementos
 * Sem eventos
 */
 type Props = {
  definition?: string,
  children?: React.Node
 }

 const props_obj ={
     definition: "string"
 }

const Column = (props: Props) => {
    validate(props, props_obj);
    var definition = (props.definition == undefined) ? "column" : "column "+ props.definition;
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    // caracter√≠sticas do bulma para a coluna
    definition : PropTypes.string
}

export default Column;