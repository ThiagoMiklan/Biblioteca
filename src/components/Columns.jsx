// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import {validate} from '../tools/type_validations.js';

/**
 *  Funciona como um container para Column
 *  Baseado na classe Columns do Bulma
 *  Sem eventos
 */
type Props = {
  definition?: string,
  children?: React.Node
 }

 const props_obj ={
     definition: 'string'
 }

const Columns = (props: Props)=>{
        validate(props, props_obj);
        var definition = (props.definition == undefined)? "columns": "columns "+ props.definition;
        return <div className={definition}>
        {
            React.Children.map(props.children, (child, i)=>{
                return child;
            })
        }
    </div>

}

Columns.propTypes = {
    definition: PropTypes.string
}



export default Columns