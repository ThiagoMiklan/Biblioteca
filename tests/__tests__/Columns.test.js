import React from 'react';
import renderer from 'react-test-renderer';
import Columns from '../../src/components/Columns';
import Column from '../../src/components/Column';


test('columns normal',()=>{
    const component = renderer.create(
         <Columns >
    <Column definition="has-background-primary">
      First column
    </Column>

    <Column definition="has-background-primary">
      Second column
    </Column>

    <Column definition="has-background-primary">
      Third column
    </Column>

    <Column definition="has-background-primary">
      Fourth column
    </Column>
  </Columns>
  ).toJSON();
    expect(component).toMatchSnapshot();
});

test('columns with name',()=>{
    const component = renderer.create(
        <>
         <Columns>
    <Column definition="is-full has-background-primary">
      full
      </Column>

  </Columns>

  <Columns>
    <Column definition="is-four-fifths has-background-primary">
      four-fifths
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-three-quarters has-background-primary">
      three-quarters
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-two-thirds has-background-primary">
      two-thirds
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-three-fifths has-background-primary">
      three-fifths
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-half has-background-primary">
      half
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-two-fifths has-background-primary">
      two-fifths
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-one-third has-background-primary">
      one-third
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-one-quarter has-background-primary">
      one-quarter
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-one-fifth has-background-primary">
      one-fifth
      </Column>
  </Columns>
  </>
  ).toJSON();
    expect(component).toMatchSnapshot();
});


test('columns with numbers',()=>{
    const component = renderer.create(
        <>
      <Columns>
    <Column definition="is-1 has-background-primary">
      1
      </Column>

  </Columns>

  <Columns>
    <Column definition="is-2 has-background-primary">
      2
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-3 has-background-primary">
      3
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-4  has-background-primary">
      4
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-5  has-background-primary">
      5
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-6  has-background-primary">
      6
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-7 has-background-primary">
      7
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-8 has-background-primary">
      8
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-9 has-background-primary">
      9
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-10 has-background-primary">
      10
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-11 has-background-primary">
      11
      </Column>
  </Columns>

  <Columns>
    <Column definition="is-12  has-background-primary">
      12
      </Column>
  </Columns>
  </>
  ).toJSON();
    expect(component).toMatchSnapshot();
});


test('columns offset',()=>{
    const component = renderer.create(
        <>
    <Columns definition="is-mobile">
    <Column definition="is-half is-offset-one-quarter has-background-primary">
      half offset-one-quarter
  </Column>
  </Columns>

  <Columns  definition="is-mobile">
    <Column definition="is-three-fifths is-offset-one-fifth has-background-primary">
      three-fifths offset-one-fifth
  </Column>
  </Columns>

  <Columns  definition="is-mobile">
    <Column definition="is-4 is-offset-8 has-background-primary">
      4 offset-8
  </Column>
  </Columns>

  <Columns definition="is-mobile">
    <Column definition="is-11 is-offset-1 has-background-primary">
      11 offset-1
  </Column>
  </Columns>
  </>
  ).toJSON();
    expect(component).toMatchSnapshot();
});


test('columns narrow',()=>{
    const component = renderer.create(
        <>
    <Columns>
    <Column definition="is-narrow">
      <p class="title is-5">Narrow column</p>
      <p class="subtitle">This column is only 200px wide.</p>
    </Column>
  </Columns>
  </>
  ).toJSON();
    expect(component).toMatchSnapshot();
});



