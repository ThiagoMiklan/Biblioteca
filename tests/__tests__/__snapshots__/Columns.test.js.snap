// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`columns narrow 1`] = `
<div
  className="columns"
>
  <div
    className="column is-narrow"
  >
    <p
      class="title is-5"
    >
      Narrow column
    </p>
    <p
      class="subtitle"
    >
      This column is only 200px wide.
    </p>
  </div>
</div>
`;

exports[`columns normal 1`] = `
<div
  className="columns"
>
  <div
    className="column has-background-primary"
  >
    First column
  </div>
  <div
    className="column has-background-primary"
  >
    Second column
  </div>
  <div
    className="column has-background-primary"
  >
    Third column
  </div>
  <div
    className="column has-background-primary"
  >
    Fourth column
  </div>
</div>
`;

exports[`columns offset 1`] = `
Array [
  <div
    className="columns is-mobile"
  >
    <div
      className="column is-half is-offset-one-quarter has-background-primary"
    >
      half offset-one-quarter
    </div>
  </div>,
  <div
    className="columns is-mobile"
  >
    <div
      className="column is-three-fifths is-offset-one-fifth has-background-primary"
    >
      three-fifths offset-one-fifth
    </div>
  </div>,
  <div
    className="columns is-mobile"
  >
    <div
      className="column is-4 is-offset-8 has-background-primary"
    >
      4 offset-8
    </div>
  </div>,
  <div
    className="columns is-mobile"
  >
    <div
      className="column is-11 is-offset-1 has-background-primary"
    >
      11 offset-1
    </div>
  </div>,
]
`;

exports[`columns with name 1`] = `
Array [
  <div
    className="columns"
  >
    <div
      className="column is-full has-background-primary"
    >
      full
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-four-fifths has-background-primary"
    >
      four-fifths
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-three-quarters has-background-primary"
    >
      three-quarters
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-two-thirds has-background-primary"
    >
      two-thirds
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-three-fifths has-background-primary"
    >
      three-fifths
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-half has-background-primary"
    >
      half
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-two-fifths has-background-primary"
    >
      two-fifths
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-one-third has-background-primary"
    >
      one-third
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-one-quarter has-background-primary"
    >
      one-quarter
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-one-fifth has-background-primary"
    >
      one-fifth
    </div>
  </div>,
]
`;

exports[`columns with numbers 1`] = `
Array [
  <div
    className="columns"
  >
    <div
      className="column is-1 has-background-primary"
    >
      1
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-2 has-background-primary"
    >
      2
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-3 has-background-primary"
    >
      3
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-4  has-background-primary"
    >
      4
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-5  has-background-primary"
    >
      5
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-6  has-background-primary"
    >
      6
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-7 has-background-primary"
    >
      7
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-8 has-background-primary"
    >
      8
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-9 has-background-primary"
    >
      9
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-10 has-background-primary"
    >
      10
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-11 has-background-primary"
    >
      11
    </div>
  </div>,
  <div
    className="columns"
  >
    <div
      className="column is-12  has-background-primary"
    >
      12
    </div>
  </div>,
]
`;
